### PR TITLE
API_Integration for books and book details

### PR DESCRIPTION
**Pull request by betymesfin**
Integrated APIs in `App.jsx` to fetch all books and in `BookDetail.jsx` to fetch books by ID.  
 Updated `BookCard.jsx` to split the published date and include `children` props.  
 Enhanced `SearchList.jsx` by adding links to book details and a "View Details" button.
Also made a little modification on `BookCard.css` to break long title.

### CODE REVIEW

**Review by olgakahraman — APPROVED** [review]
👍great job!

File: src/App.jsx [hunk 1]
```diff
@@ -21,12 +56,16 @@ function App() {
...
 						path="/find-book"
-						element={<FindABook />}
+						element={<FindABook booksData={books} />}
 					/>
 					<Route
 						path="/"
-						element={<Home booksData={booksData} />}
+						element={<Home booksData={books} />}
 					/>
+          <Route
```
Comment (ElijahMSmith) [inline src/App.jsx]: Make sure to run your formatter!

File: src/Pages/home.jsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ function Home({ booksData }) {
 		};
 		pickRandomBook();
 
-		const interval = setInterval(pickRandomBook, 5000);
+		const interval = setInterval(pickRandomBook, 50000000);
```
Comment (ElijahMSmith) [inline src/Pages/home.jsx]: Is this an intentional change?

File: src/components/BookDetails/BookDetails.jsx [hunk 3]
```diff
@@ -1,41 +1,62 @@
...
+
+    fetchBookDetails();
+  }, [id]);
+
+  if (loading) return <h2>Loading...</h2>;
+  if (error) return <h2>Error: {error}</h2>;
+  if (!book) return <h2>Book not found.</h2>;
+
+ 
 	return (
```
Comment (ElijahMSmith) [inline src/components/BookDetails/BookDetails.jsx:35]: Formatter here too!

**Review by ElijahMSmith — APPROVED** [review]
This is excellent work!

**Comment by ElijahMSmith** [comment]
One thing to note that I forgot, we should avoid hard-coding our backend URL into the application. This makes it hard for someone on the team to swap out the URL for `localhost` if they are running a development version of the backend on their machine.

Instead of `https://hh-team3-back.onrender.com/api/whatever`, let's use an environment variable to construct the URL
`process.meta.env.BASE_URL + "/api/whatever".

https://vite.dev/guide/env-and-mode

It would be good to leave a comment in the frontend team channel that you added this environment variable when you merge in this code so the other team members can add it to their local `.env` files.

File: src/Pages/home.jsx (hunk 2 above)
Comment (betymesfin) [reply src/Pages/home.jsx]: Sorry , I forgot to change the interval back to the original value. I had to increase it to make debugging easier in the developer console.

**Pull request merged**